src/scripts/fetchRouteConfigs.js

```diff
@@ -0,0 +1,207 @@
+            // load the json file from mentioned location
+            routeJson.data = await loadJsonFile(url);
+            
+            routeJson.success = routeJson?.data.routes.length > 0  && isValidJson(routeJson) ? true : false
+        }else{
+            routeJson.success = false
+        }
+        let fileData = {
+            routes: []
+        }
+        let orchestratedFileData = {
+            routes: []
+        }
+        let outputFile = ""
+        if (routeJson.success) {
+            // extract base package name to create json file 
+            outputFile = path.join(outputDir, routeJson.data.routes[0]?.targetPackages[0]?.packageName + '.json')
+
+            routeJson.data.routes.map(async (routeDetails) => {
+                if (!routeDetails.orchestrated) {
+                    // if the api is not orchestrated , write it in the package specific file 
+                    if(!isADuplicateRoute(routeDetails, fileData.routes)) fileData.routes.push(routeDetails)
+                }
+                else {
+                    orchestratedFileData.routes.push(routeDetails)
+                }
+            })
+            // Ensure the directory exists
+            const dir = path.dirname(outputDir);
+            await fs.mkdir(dir, { recursive: true });
+
+            // Write the JSON file (creates or overwrites the file)
+            await fs.writeFile(outputFile, JSON.stringify(fileData, null, 4));
+
+            if(orchestratedFileData.routes.length > 0) await handleOrchestratedFile(orchestratedFileData.routes)
+
+            console.log(`File generated : ' ${outputFile} '.`);
+            if(orchestratedFileUpdatedFlag) console.log(`Orchestrated File Updated for package : ' ${outputFile.split('.json')[0]} '.`)
+            orchestratedFileUpdatedFlag = false
+        }
+        
+    } catch (error) {
+        console.log(error)
+    }
+
+}
+/**
+ * Handles orchestrated calls, opens the orchestrated.json file and w
+ * @param {string} url - The URL to fetch.
+ */
+const handleOrchestratedFile = async (newRoutes) => {
+    try {
+        const existingRoutes = await fs.readFile(orchestratedFile, 'utf8')
+        let fileWriteFlag = false
+        let routes = []
+        try{
+            routes = JSON.parse(existingRoutes).routes || []
+            console.log("Existing routes : " , routes.length)
+        }catch(error){
+            routes = []
+        }
+
+        newRoutes.map(async (routeDetails) => {
+            if (!isADuplicateRoute(routeDetails, routes)) { 
+                routes.push(routeDetails)
+                fileWriteFlag = true
+            }
+        })
+
+        if(fileWriteFlag) await fs.writeFile(orchestratedFile, JSON.stringify({routes}, null, 4));
+
+    } catch (error) {
+        if (error.code === 'ENOENT') {
+            // if the file is not created , create
+            await fs.writeFile(orchestratedFile, JSON.stringify({ routes: newRoutes }, null, 4));
```

Comment:
No, for the first time the routsConfig folder will be empty , then while opening it will throw ENOENT error. 
At that time we have to create the file and add the new routes. Else if the file is already there , we have to open the file take the existing routes and push new routes to it. @priyanka-TL